client/src/document/hooks.ts

```diff
@@ -131,10 +131,40 @@ function codeForHeading(
       })
       .join("\n");
     if (src) {
-      empty = false;
       code[part] += src;
     }
   }
+  return nodes.length ? { code, nodes } : null;
+}
+
+function getLanguage(node: Element): string | null {
+  for (const part of LIVE_SAMPLE_PARTS) {
+    if (node.classList.contains(part)) {
+      return part;
+    }
+  }
+  return null;
+}
+
+function getCodeAndNodesForIframeBySampleClass(cls: string, src: string) {
```

Comment:
These are the newfangled macro-less live samples you were discussing on slack? We'll need some form of documentation somewhere for them.